src/isomorphic/classic/types/checkPropTypes.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule checkPropTypes
+ */
+
+'use strict';
+
+var ReactPropTypesSecret = require('ReactPropTypesSecret');
+
+var invariant = require('invariant');
+var warning = require('warning');
+
+var loggedTypeFailures = {};
+
+/**
+ * Assert that the values match with the type specs.
+ * Error messages are memorized and will only be shown once.
+ *
+ * @param {object} typeSpecs Map of name to a ReactPropType
+ * @param {object} values Runtime values that need to be type-checked
+ * @param {string} location e.g. "prop", "context", "child context"
+ * @param {string} componentName Name of the component for error messages.
+ * @param {?Function} formatMessage Function that transforms the error message, to add additional info.
+ * @param {?boolean} warnOnRepeat Whether or not repeated warnings should be skipped.
+ * @private
+ */
+function checkPropTypes(typeSpecs, values, location, componentName, formatMessage, warnOnRepeat) {
```

Comment:
Seemed like it might be a useful opt-in, perhaps if a lib's error messages are less specific and don't include the component name. But yeah I'll just get rid of it.